sci-libs/mlpack/mlpack-3.0.0.recipe

```diff
@@ -0,0 +1,200 @@
+	cmd:mlpack_hmm_train = $portVersion compat >= 3
+	cmd:mlpack_hmm_viterbi = $portVersion compat >= 3
+	cmd:mlpack_hoeffding_tree = $portVersion compat >= 3
+	cmd:mlpack_kernel_pca = $portVersion compat >= 3
+	cmd:mlpack_kfn = $portVersion compat >= 3
+	cmd:mlpack_kmeans = $portVersion compat >= 3
+	cmd:mlpack_knn = $portVersion compat >= 3
+	cmd:mlpack_krann = $portVersion compat >= 3
+	cmd:mlpack_lars = $portVersion compat >= 3
+	cmd:mlpack_linear_regression = $portVersion compat >= 3
+	cmd:mlpack_local_coordinate_coding = $portVersion compat >= 3
+	cmd:mlpack_logistic_regression = $portVersion compat >= 3
+	cmd:mlpack_lsh = $portVersion compat >= 3
+	cmd:mlpack_mean_shift = $portVersion compat >= 3
+	cmd:mlpack_nbc = $portVersion compat >= 3
+	cmd:mlpack_nca = $portVersion compat >= 3
+	cmd:mlpack_nmf = $portVersion compat >= 3
+	cmd:mlpack_pca = $portVersion compat >= 3
+	cmd:mlpack_perceptron = $portVersion compat >= 3
+	cmd:mlpack_preprocess_binarize = $portVersion compat >= 3
+	cmd:mlpack_preprocess_describe = $portVersion compat >= 3
+	cmd:mlpack_preprocess_imputer = $portVersion compat >= 3
+	cmd:mlpack_preprocess_split = $portVersion compat >= 3
+	cmd:mlpack_radical = $portVersion compat >= 3
+	cmd:mlpack_random_forest = $portVersion compat >= 3
+	cmd:mlpack_range_search = $portVersion compat >= 3
+	cmd:mlpack_softmax_regression = $portVersion compat >= 3
+	cmd:mlpack_sparse_coding = $portVersion compat >= 3
+	"
+REQUIRES_cli="
+	mlpack$secondaryArchSuffix == $portVersion base
+	$REQUIRES
+	"
+
+PROVIDES_devel="
+	mlpack${secondaryArchSuffix}_devel = $portVersion compat >= 3
+	devel:libmlpack$secondaryArchSuffix = $portVersion compat >= 3
+	"
+REQUIRES_devel="
+	mlpack$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libarmadillo$secondaryArchSuffix
+	devel:libboost_program_options$secondaryArchSuffix
+	devel:libboost_serialization$secondaryArchSuffix
+	devel:libboost_system$secondaryArchSuffix
+	devel:libboost_unit_test_framework$secondaryArchSuffix
+	devel:libhdf5$secondaryArchSuffix
+	devel:liblapack$secondaryArchSuffix
+	devel:libopenblas$secondaryArchSuffix
+	devel:libsuperlu$secondaryArchSuffix
+	devel:libxml2$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage mlpack$secondaryArchSuffix \
+	$libDir/libmlpack.so.3.0
+
+BUILD()
+{
+	mkdir -p build && cd "$_"
+
+	cmake .. \
+		-DCMAKE_INSTALL_PREFIX=$prefix \
+		-DBUILD_TESTS=OFF \
+		-DDEBUG=OFF \
+		-DPROFILE=OFF
+
+	# mlpack easily eats up all your ram, so don't do parallel jobs!
+	make
+}
+
+INSTALL()
+{
+	cd build
+
+	make install $jobArgs
+
+	mkdir -p $includeDir
+	mv $prefix/include/* $includeDir/
+	rmdir $prefix/include
+
+	prepareInstalledDevelLib libmlpack
+
+	fixPkgconfig
+
+	packageEntries cli \
+		$binDir/mlpack_adaboost \
+		$binDir/mlpack_approx_kfn \
+		$binDir/mlpack_cf \
```

Comment:
Since all the `cmd:` entries are in `PROVIDES_cli`, you don't need to list them here.
You can therefore replace the full list by`"$binDir"` after  `packageEntries cli`.
```diff
 	packageEntries cli \
+		"$binDir"
-		$binDir/mlpack_adaboost \
-		$binDir/mlpack_approx_kfn \
-		$binDir/mlpack_cf \
```
...